misc-scripts/xref_mapping/XrefParser/HPAParser.pm

```diff
@@ -39,70 +42,55 @@ sub run {
   my $source_id    = $ref_arg->{source_id};
   my $species_id   = $ref_arg->{species_id};
   my $files        = $ref_arg->{files};
-  my $verbose      = $ref_arg->{verbose};
-  my $dbi          = $ref_arg->{dbi};
-  $dbi = $self->dbi unless defined $dbi;
+  my $verbose      = $ref_arg->{verbose} || 0;
+  my $dbi          = $ref_arg->{dbi} || $self->dbi;
 
-  if((!defined $source_id) or (!defined $species_id) or (!defined $files) ){
-    croak "Need to pass source_id, species_id, files and rel_file as pairs";
-  }
-  $verbose |=0;
+  croak "Need to pass source_id, species_id, files and rel_file as pairs"
+    unless defined $source_id and defined $species_id and defined $files;
 
-  my $filename = @{$files}[0];
+  my $file = @{$files}[0];
 
-  my $file_io = $self->get_filehandle($filename);
-  if ( !defined($file_io) ) {
-    return 1;
-  }
+  my $file_io = $self->get_filehandle($file);
+  croak "ERROR: Could not open $file\n" unless defined $file_io;
 
-  my $parsed_count = 0;
+  my $input_file = Text::CSV->new({ sep_char           => ",",
+				    empty_is_undef     => 1
+				  }) or croak "Cannot use file $file: ".Text::CSV->error_diag ();
+  
+  $file_io->getline(); # skip header # Antibody,antibody_id,ensembl_peptide_id,link
 
-  $file_io->getline();
-
-  while ( defined( my $line = $file_io->getline() ) ) {
-
-    $line =~ s/\s*$//;
-		
-
-    my ( $antibody, $antibody_id, $ensembl_peptide_id, $link) = split( /,/, $line );
+  my $parsed_count = 0;
+  while ( my $row = $input_file->getline( $file_io ) ) {
+    map { $_ =~ s/\s*$// } @{$row};
```

Comment:
maps with no return are a perlcritic warning. You could just feed these into the variables defined below if you must map{}.

Reply:
They would all be set to 1, which is what the expression inside map evaluates to. I have to be boring by removing map and make explicit statements for each one of the declared variables.

Reply:
Now with explicit statements.